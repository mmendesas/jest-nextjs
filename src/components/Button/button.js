import React from 'react';
import { func } from 'prop-types'

function Button({ onClick, children, ...props }) {
  const handleClick = () => {
    onClick('hit button');
  }

  return (
    <button id="btn123" data-testid='btn123' onClick={handleClick} {...props}>
      {children}
    </button>
  );
}

Button.defaultProps = {
  onClick: () => { }
}

Button.propTypes = {
  onClick: func
}

export default Button;