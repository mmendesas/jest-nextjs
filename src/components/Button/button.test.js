
import React from 'react'
import { render, fireEvent } from '@testing-library/react';
import Button from './button';
import 'jest-styled-components';

describe('[Component] Button', () => {

  it('should render correctly [snapshot]', async () => {
    // arrange
    // act
    // assert
  });


  it('should call function on click', async () => {
    // arrange
    // act
    // assert
  });

  it('should call default function if onClick is not defined', async () => {
    // arrange
    // act
    // assert
  });

});