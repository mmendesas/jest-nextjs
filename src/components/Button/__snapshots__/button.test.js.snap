// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] Button should render correctly [snapshot] 1`] = `
<button
  data-testid="btn123"
  id="grosa"
>
  click me
</button>
`;
